src/engines/kotor/gui/chargen/classselection.cpp

```diff
@@ -0,0 +1,133 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  The class selection menu.
+ */
+
+#include "src/aurora/talkman.h"
+#include "src/engines/kotor/gui/chargen/classselection.h"
+
+#include "src/engines/kotor/gui/widgets/kotorwidget.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+ClassSelectionMenu::ClassSelectionMenu(Module *module, ::Engines::Console *console) : GUI(console),
+	_module(module) {
+
+	load("classsel");
+
+	addBackground(kBackgroundTypeMenu);
+
+	// get the six class buttons
+	_maleScoundrel = getButton("BTN_SEL1");
+	_maleScout = getButton("BTN_SEL2");
+	_maleSoldier = getButton("BTN_SEL3");
+	_femaleSoldier = getButton("BTN_SEL4");
+	_femaleScout = getButton("BTN_SEL5");
+	_femaleScoundrel = getButton("BTN_SEL6");
+
+	// get the description label
+	_labelDesc = getLabel("LBL_DESC");
+	_labelDesc->setWrapped(true);
+	_labelDesc->setText(TalkMan.getString(32111));
+
+	// get the title label
+	_labelTitle = getLabel("LBL_CLASS");
+	_labelTitle->setText("");
+
+	// get the class descriptions
+	_soldierDesc = TalkMan.getString(32111);
+	_scoutDesc = TalkMan.getString(32110);
+	_scoundrelDesc = TalkMan.getString(32109);
+
+	//combine the titles with gender prefix and class name
+	Common::UString malePrefix, femalePrefix;
+	malePrefix = TalkMan.getString(358);
+	femalePrefix = TalkMan.getString(359);
+	_soldierMaleTitle = malePrefix + " " + TalkMan.getString(134);
+	_soldierFemaleTitle = femalePrefix + " " + TalkMan.getString(134);
+	_scoutMaleTitle = malePrefix + " " + TalkMan.getString(133);
+	_scoutFemaleTitle = femalePrefix + " " + TalkMan.getString(133);
+	_scoundrelMaleTitle = malePrefix + " " + TalkMan.getString(135);
+	_scoundrelFemaleTitle = femalePrefix + " " + TalkMan.getString(135);
+}
+
+void ClassSelectionMenu::callbackRun() {
```

Comment:
Hmm, doesn't that set/change the text every time the callback is called? I think it would be best if you remembered over which button the mouse is hovering and only set the text when the selection changes.